vulnerabilities/throttling.py

```diff
@@ -22,3 +22,42 @@ def allow_request(self, request, view):
             return True
 
         return super().allow_request(request, view)
+
+    def get_cache_key(self, request, view):
+        """
+        Return the cache key to use for this request.
+        """
+        if request.user.is_authenticated:
+            ident = request.user.pk
+        else:
+            ident = self.get_ident(request)
+
+        return self.cache_format % {"scope": self.scope, "ident": ident}
+
+
+class VulnerablePackagesAPIThrottle(StaffUserRateThrottle):
+    scope = "vulnerable_packages"
+
+
+class BulkSearchPackagesAPIThrottle(StaffUserRateThrottle):
+    scope = "bulk_search_packages"
+
+
+class PackagesAPIThrottle(StaffUserRateThrottle):
+    scope = "packages"
+
+
+class VulnerabilitiesAPIThrottle(StaffUserRateThrottle):
+    scope = "vulnerabilities"
+
+
+class AliasesAPIThrottle(StaffUserRateThrottle):
+    scope = "aliases"
+
+
+class CPEAPIThrottle(StaffUserRateThrottle):
+    scope = "cpes"
+
+
+class BulkSearchCPEAPIThrottle(StaffUserRateThrottle):
+    scope = "bulk_search_cpes"
```

Comment:
There's no better way to do this? It looks like a lot of duplication here.

Reply:
I have now used ScopeRatedThrottling, please see it looks good